mu-plugins/blocks/time/index.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Block Name: Time
+ * Description: Attempts to parse a time string like <code>[time]any-valid-time-string-here[/time]</code> and creates a format that shows it in the viewers local time zone.
+ *
+ * @package wporg
+ */
+
+namespace WordPressdotorg\MU_Plugins\Time;
+
+use function WordPressdotorg\MU_Plugins\Helpers\register_assets_from_metadata;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Enqueues the script for the Time format in the block editor using the metadata loaded from the `block.json` file,
+ * and adds the filter for transforming the time blocks.
+ */
+function init() {
+	// Add the JS script to add the Time formatting option to the toolbar. The dependencies are autogenerated in block.json,
+	// and can be read with `wp_json_file_decode` & `register_block_script_handle.
+	$metadata = get_metadata();
+	$editor_script_handle = register_block_script_handle( $metadata, 'editorScript', 0 );
+	add_action(
+		'enqueue_block_assets',
+		function() use ( $editor_script_handle ) {
+			if ( wp_should_load_block_editor_scripts_and_styles() && is_admin() ) {
+				wp_enqueue_script( $editor_script_handle );
+			}
+		}
+	);
+
+	add_filter( 'the_content', __NAMESPACE__ . '\transform_times', 99, 1 );
+}
+
+/**
+ * Reads the block.json file and returns the metadata.
+ */
+function get_metadata() {
+	$metadata_file = __DIR__ . '/build/block.json';
+	$metadata = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
+	$metadata['file'] = $metadata_file;
+
+	return $metadata;
+}
+
+/**
+ * Builds the time block output.
+ *
+ * This implements replacing the raw time strings in the post content with formatted times able to be converted to local times with JS.
+ *
+ * @param string $content Post content.
+ * @return string Content with display times reformatted.
+ */
+function transform_times( $content ) {
+	if ( empty( $content || is_admin() ) ) {
+		return $content;
+	}
+
+	// Find the time block elements by the classname "wporg-time"
+	$dom = new \DOMDocument();
+
+	// Ignore warnings about htlm5 tags.
+	$dom->loadHTML( $content, LIBXML_NOERROR );
```

Comment:
Without this there can be **many** warnings displayed (nav, time, and many other html5 tags aren't supported).

Alternatively we could use regexp or a different dom parser...